google/cloud/bigtable/tests/table_sample_rows_integration_test.cc

```diff
@@ -0,0 +1,136 @@
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+namespace {
+
+void VerifySamples(StatusOr<std::vector<RowKeySample>> samples) {
+  ASSERT_STATUS_OK(samples);
+
+  // It is somewhat hard to verify that the values returned here are correct.
+  // We cannot check the specific values, not even the format, of the row keys
+  // because Cloud Bigtable might return an empty row key (for "end of table"),
+  // and it might return row keys that have never been written to.
+  // All we can check is that this is not empty, and that the offsets are in
+  // ascending order.
+  EXPECT_FALSE(samples->empty());
+  std::int64_t previous = 0;
+  for (auto const& s : *samples) {
+    EXPECT_LE(previous, s.offset_bytes);
+    previous = s.offset_bytes;
+  }
+  // At least one of the samples should have non-zero offset:
+  auto last = samples->back();
+  EXPECT_LT(0, last.offset_bytes);
+}
+
+using testing::TableTestEnvironment;
+
+class SampleRowsIntegrationTest
+    : public ::google::cloud::testing_util::IntegrationTest {
+ public:
+  static void SetUpTestSuite() {
+    table_ = absl::make_unique<Table>(
+        bigtable::Table(bigtable::CreateDefaultDataClient(
+                            TableTestEnvironment::project_id(),
+                            TableTestEnvironment::instance_id(),
+                            ClientOptions().disable_tracing("rpc")),
+                        TableTestEnvironment::table_id()));
+
+    int constexpr kBatchCount = 10;
+    int constexpr kBatchSize = 5000;
+    int constexpr kColumnCount = 10;
+    std::string const family = "family1";
+    int rowid = 0;
+
+    for (int batch = 0; batch != kBatchCount; ++batch) {
+      BulkMutation bulk;
+      for (int row = 0; row != kBatchSize; ++row) {
+        std::ostringstream os;
+        os << "row:" << std::setw(9) << std::setfill('0') << rowid;
+
+        // Build a mutation that creates 10 columns.
+        SingleRowMutation mutation(os.str());
+        for (int col = 0; col != kColumnCount; ++col) {
+          std::string column_id = "c" + std::to_string(col);
+          std::string value = column_id + "#" + os.str();
+          mutation.emplace_back(SetCell(family, std::move(column_id),
+                                        std::chrono::milliseconds(0),
+                                        std::move(value)));
+        }
+        bulk.emplace_back(std::move(mutation));
+        ++rowid;
+      }
+      auto failures = table_->BulkApply(std::move(bulk));
+      for (auto const& failure : failures) {
+        std::cout << failure.original_index() << ": " << failure.status()
+                  << "\n";
+      }
+      ASSERT_THAT(failures, ::testing::IsEmpty());
+    }
+  }
+
+  static void TearDownTestSuite() { table_ = nullptr; }
+
+ protected:
+  static std::unique_ptr<Table> table_;
+};
+
+std::unique_ptr<Table> SampleRowsIntegrationTest::table_;
```

Comment:
I copied [this](https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/spanner/integration_tests/data_types_integration_test.cc#L88) but without understanding. Maybe someone could explain over chat what is happening